### PR TITLE
LIPS updates according to merge-pack upgrade

### PR DESCRIPTION
**Pull request by loga4**
Updated LIP specifications according to merge-ready-pack 

context: https://research.lido.fi/t/announcement-merge-ready-protocol-service-pack/2184

### CODE REVIEW

File: LIPS/lip-12.md [hunk 1]
```diff
@@ -56,68 +56,149 @@ A rewards vault is represented with the new `LidoMevTxFeeVault` contract. Reward
...
 In addition to the existing `pushBeacon` code, claims EL-collected rewards after saving current beacon balance, and in case of positive beacon chain delta increase overall rewards amount by the EL-collected value before calling `distributeRewards` function.
-* Must not change rewards distribution if there are no EL-collected rewards or `LidoMevTxFeeVault` address is zero.
+* Must not change rewards distribution if there are no EL-collected rewards or `LidoExecutionLayerRewardsVault` address is zero.
 * Must not distribute any rewards if beacon chain delta (with a correction by newly appeared validators amount) is zero or negative.
 * Must not introduce additional revert reasons if all permissions agree.
 
-The function is also renamed to better match it's new fucntionality.
+The function is also renamed to better match it's new functionality.
 
 #### Function: getTotalMevTxFeeCollected
```
Comment (folkyatina) [inline LIPS/lip-12.md]: Remove?

Comment (folkyatina) [reply LIPS/lip-12.md]: And `getTotalMevTxFeeCollected()`?

File: LIPS/lip-14.md [hunk 2]
```diff
@@ -18,7 +18,8 @@ Add a staking rate limiting feature to have the soft moving cap for the stake am
 We propose to implement a rate limiting mechanics by using the following conventions:
 
 - Staking can be paused completely by calling the `pauseStaking()` method
-- Staking can be resumed then with a rate limit set or without rate limit at all
+- Staking can be resumed then by calling the `resumeStaking()` method 
+- Staking limit can be removed by calling the `removeStakingLimit()` method
 - The rate limit is managed by two parameters:
```
Comment (folkyatina) [inline LIPS/lip-14.md]: I'd rather go with three points here. 
- Staking resume/pause. 
- Staking limit set/remove
- Staking limit parameters

File: LIPS/lip-14.md [hunk 3]
```diff
@@ -45,41 +46,92 @@ We propose to accomplish the `Lido.sol` contract with additional functions to co
...
-function calculateCurrentStakeLimit() external view returns (
+function getCurrentStakeLimit() public view returns (uint256)
+```
+Returns how much Ether can be staked in the current block
+- 2^256 - 1 if staking is unlimited;
+- 0 if staking is paused or if limit is exhausted.
+
+#### Function: getStakeLimitFullInfo
+```solidity
+function getStakeLimitFullInfo() external view returns (
```
Comment (folkyatina) [inline LIPS/lip-14.md:101]: I think we need a description for calculated parameters here as well as a brief motivation why did we choose exactly these parameters.

File: LIPS/lip-7.md [hunk 4]
```diff
@@ -131,15 +132,15 @@ Iteratively calls `callback.processLidoOracleReport` for each stored `callback`
 
 ## Gas effects
 
-In the case of `ComositePostRebaseBeaconReceiver` used with only one `IBeaconReportReceiver` callback set, spendings increase approximately by the ~4600 gas compared to direct usage of the underlying callback with `LidoOracle` (without composite adapter).
+In the case of `CompositePostRebaseBeaconReceiver` used with only one `IBeaconReportReceiver` callback set, spending's increase approximately by the ~4600 gas compared to direct usage of the underlying callback with `LidoOracle` (without composite adapter).
 
 ## Security considerations
 
-#### Upgradability
+#### Upgradeability
```
Comment (folkyatina) [inline LIPS/lip-7.md:139]: Actually, both are right. )

File: LIPS/lip-7.md [hunk 5]
```diff
@@ -131,15 +132,15 @@ Iteratively calls `callback.processLidoOracleReport` for each stored `callback`
 
 ## Gas effects
 
-In the case of `ComositePostRebaseBeaconReceiver` used with only one `IBeaconReportReceiver` callback set, spendings increase approximately by the ~4600 gas compared to direct usage of the underlying callback with `LidoOracle` (without composite adapter).
+In the case of `CompositePostRebaseBeaconReceiver` used with only one `IBeaconReportReceiver` callback set, spending's increase approximately by the ~4600 gas compared to direct usage of the underlying callback with `LidoOracle` (without composite adapter).
```
Comment (folkyatina) [inline LIPS/lip-7.md]: Replace `spending's` with `spending`?

**Review by folkyatina — COMMENTED** [review]
Couple of places to improve

File: LIPS/lip-12.md (hunk 1 above)
Comment (loga4) [reply LIPS/lip-12.md]: yep, and `getMevTxFeeVault` too, ty!

File: LIPS/lip-14.md (hunk 3 above)
Comment (loga4) [reply LIPS/lip-14.md:101]: As for the calculated variables, this is redundant here. There is a detailed description above, plus a discussion on the forum. According to the parameters, I added a small note about integrations)

File: LIPS/lip-14.md (hunk 2 above)
Comment (loga4) [reply LIPS/lip-14.md]: Agree, it looks much cleaner!

File: LIPS/lip-7.md (hunk 4 above)
Comment (loga4) [reply LIPS/lip-7.md:139]: ok)

File: LIPS/lip-7.md (hunk 5 above)
Comment (loga4) [reply LIPS/lip-7.md]: done

**Review by TheDZhon — APPROVED** [review]
Fasten seat belts :rocket:

**Pull request merged**